kenjutsu/kenjutsu.py

```diff
@@ -53,64 +53,63 @@ def reformat_slice(a_slice, a_length=None):
-        new_slice_start = 0
-
-    new_slice_stop = a_slice.stop
+    start = a_slice.start
+    stop = a_slice.stop
+    step = a_slice.step
 
+    # Fill unknown values.
+    if step is None:
+        step = 1
+    if start is None:
+        if step > 0:
+            start = 0
+        elif step < 0:
+            start = -1
+    if (stop is None) and (step > 0):
+        stop = a_length
 
     # Make adjustments for length
-
     if a_length is not None:
-        if (new_slice_step < -a_length):
-            new_slice_step = -a_length
-        elif (new_slice_step > a_length):
-            new_slice_step = a_length
-
-        if (new_slice_start is None) and (new_slice_step < 0):
-            new_slice_start = a_length
-        elif (new_slice_start <= -a_length) and (new_slice_step > 0):
-            new_slice_start = 0
-        elif (new_slice_start < -a_length) and (new_slice_step < 0):
-            new_slice_start = new_slice_stop = 0
-        elif (new_slice_start > a_length) and (new_slice_step > 0):
-            new_slice_start = a_length
-        elif (new_slice_start > a_length) and (new_slice_step < 0):
-            new_slice_start = a_length
-        elif (new_slice_start < 0) and (new_slice_step > 0):
-            new_slice_start += a_length
-        elif (new_slice_start < 0) and (new_slice_step < 0):
-            new_slice_start += a_length
-
-        if (new_slice_stop is None) and (new_slice_step > 0):
-            new_slice_stop = a_length
-        elif (new_slice_stop is None) and (new_slice_step < 0):
-            pass
-        elif (new_slice_stop <= -a_length) and (new_slice_step > 0):
-            new_slice_stop = 0
-        elif (new_slice_stop < -a_length) and (new_slice_step < 0):
-            new_slice_stop = None
-        elif (new_slice_stop < 0) and (new_slice_step > 0):
-            new_slice_stop += a_length
-        elif (new_slice_stop < 0) and (new_slice_step < 0):
-            new_slice_stop += a_length
-        elif (new_slice_stop > a_length) and (new_slice_step > 0):
-            new_slice_stop = a_length
-        elif (new_slice_stop >= a_length) and (new_slice_step < 0):
-            new_slice_start = new_slice_stop = 0
-
-
-    # Build new slice and return.
-
-    new_slice = slice(new_slice_start, new_slice_stop, new_slice_step)
+        # Normalize out-of-bound step sizes.
+        if step < -a_length:
+            step = -a_length
+        elif step > a_length:
+            step = a_length
+
+        # Normalize bounded negative values.
+        if -a_length <= start < 0:
+            start += a_length
+        if -a_length <= stop < 0:
+            stop += a_length
+
+        # Handle out-of-bound limits.
+        if step > 0:
+            if (start > a_length) or (stop < -a_length):
+                start = stop = 0
+            elif start < -a_length:
+                start = 0
+            elif stop > a_length:
+                stop = a_length
```

Comment:
These two `elif`s should have been two `if`s as both can be true and that is ok. However, as an optimization we can put both in an `else` as clearly if first `if` is true (even with the mutation), these won't be. Submitting a PR to fix it.